connect/mirror/src/test/java/org/apache/kafka/connect/mirror/clients/admin/FakeForwardingAdminWithLocalMetadata.java

```diff
@@ -38,16 +38,13 @@
 import java.util.Collection;
 import java.util.Map;
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.TimeoutException;
 
 /** Customised ForwardingAdmin for testing only.
  * The class create/alter topics, partitions and ACLs in Kafka then store metadata in {@link FakeLocalMetadataStore}.
  * */
 
 public class FakeForwardingAdminWithLocalMetadata extends ForwardingAdmin {
     private static final Logger log = LoggerFactory.getLogger(FakeForwardingAdminWithLocalMetadata.class);
-    private final long timeout = 1000L;
```

Comment:
Any reason why removing the timeout completely instead of increasing it to some abundantly high number of seconds?
My time at Akka tells me: never wait forever; always set a timeout, even if it's high.
What do you think about putting something like 15 seconds? This way if something goes wrong and the future never finished we won't be hanging there forever.

Reply:
I didn't just raise the timeout because the MirrorSourceTask that makes these calls does not have a timeout that I could predictably meet. The task appears to either use `get()` or `whenComplete` on the futures returned by these calls.

I think your critique is correct though, and I've rewritten this to use whenComplete() to get the results of the futures, instead of get().

Reply:
I like the `whenComplete` constructs, I think they are more elegant as well.